tests/test_store/test_core.py

```diff
@@ -56,10 +94,19 @@ async def test_make_store_path_store_path(
     assert Path(store_path.store.root) == Path(tmpdir)
     path_normalized = normalize_path(path)
     assert store_path.path == (store_like / path_normalized).path
-
     assert store_path.read_only == ro
 
 
+async def test_store_path_invalid_mode_raises(tmpdir: LEGACY_PATH) -> None:
+    """
+    Test that ValueErrors are raise for invalid mode.
+    """
+    with pytest.raises(ValueError):
+        await StorePath.open(LocalStore(str(tmpdir), read_only=True), path=None, mode="w")
+    with pytest.raises(ValueError):
+        await StorePath.open(LocalStore(str(tmpdir), read_only=False), path=None, mode="x")
```

Comment:
lets parametrize the test over `mode` instead of repeating nearly identical checks

Reply:
looks like we would need to parametrize over `(read_only, mode)` tuples